### DISCUSSION TITLE
HLIL(or BNIL) instruction deletion

### DISCUSSION THREAD

**Discussion by mostobriv** [Vector35/binaryninja-api#3888] [API] [answered]
Is there any way to delete BNIL instruction directly?

**Comment by psifertex** [comment, marked as answer]
Sorry for the delay in responding. There is no way to directly delete a BNIL instruction. Here are your options:

1) Delete the underlying assembly -- the UI/API even can show you the exact native instructions that align with an HLIL instruction. Specifically the ".llils" instruction will map to the set of LLIL instructions that all have distinct native addresses.
2) Create a [workflow](https://api.binary.ninja/binaryninja.workflow-module.html). A workflow can technically remove IL instructions but they're not super easy to work with right now. For example, the [inliner](https://github.com/Vector35/binaryninja-api/blob/dev/examples/workflows/inliner/inliner.cpp) workflow shows how to replace one IL instruction with another. This can potentially be used to replace multiple instructions such that you could remove an instruction.

**Reply by mostobriv** [reply]
Thanks for your answer!
